src/common.browser/FileAudioSource.ts

```diff
@@ -141,46 +145,48 @@ export class FileAudioSource implements IAudioSource {
 
     private readHeader = (): Promise<AudioStreamFormatImpl> => {
         // Read the wave header.
-        const header: Blob = this.privFile.slice(0, 44);
-        const headerReader: FileReader = new FileReader();
-
         const headerResult: Deferred<AudioStreamFormatImpl> = new Deferred<AudioStreamFormatImpl>();
 
-        const processHeader = (event: Event): void => {
-            const header: ArrayBuffer = (event.target as FileReader).result as ArrayBuffer;
+        if (typeof (FileReader) !== "undefined") {
```

Comment:
>if (typeof (FileReader) !== "undefined") [](start = 8, length = 40)

Where did this happen, and if we can't read the header won't this just leave the promise permanently not resolved? Should it be rejected?

Reply:
Reverted to original flow